### PR TITLE
Add transformers-like api doc

### PR DESCRIPTION
**Pull request by Kaihui-intel**
## Type of Change

documentation 

## Description

detail description

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: docs/source/3x/transformers_like_api.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+4. [Usage For Intel GPU](#usage-for-intel-gpu)
+
+5. [Examples](#examples)
+
+## Introduction
+
+Transformers-like API provides a seamless user experience of model compressions on Transformer-based models by extending [Hugging Face transformers](https://github.com/huggingface/transformers) APIs, leveraging [Intel® Neural Compressor](https://github.com/intel/neural-compressor), and replacing Linear operator with [Intel® Extension for PyTorch](https://github.com/intel/intel-extension-for-pytorch).
+## Supported Algorithms
+
+| Support Device |  Rtn  |  Awq  |  Teq |  GPTQ  | AutoRound |
```
Comment (thuang6) [inline docs/source/3x/transformers_like_api.md:19]: to make it consistent with other docs, suggest change Rtn -> RTN and Awq -> AWQ, and Teq -> TEQ

File: docs/source/3x/transformers_like_api.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+3. [Usage For Intel CPU](#usage-for-cpu)
+
+4. [Usage For Intel GPU](#usage-for-intel-gpu)
+
+5. [Examples](#examples)
+
+## Introduction
+
+Transformers-like API provides a seamless user experience of model compressions on Transformer-based models by extending [Hugging Face transformers](https://github.com/huggingface/transformers) APIs, leveraging [Intel® Neural Compressor](https://github.com/intel/neural-compressor), and replacing Linear operator with [Intel® Extension for PyTorch](https://github.com/intel/intel-extension-for-pytorch).
```
Comment (thuang6) [inline docs/source/3x/transformers_like_api.md:16]: leveraging [Intel® Neural Compressor](https://github.com/intel/neural-compressor)  is inappropriate,  Transformer-like API is part of INC.  Suggest change to： 
Transformers-like API provides a seamless user experience of model compressions on Transformer-based models by extending [Hugging Face transformers](https://github.com/huggingface/transformers) APIs, leveraging neural compressor existing weight-only quantization capability and replacing Linear operator with [Intel® Extension for PyTorch](https://github.com/intel/intel-extension-for-pytorch).

File: docs/source/3x/transformers_like_api.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+# save quant model
+saved_dir = "SAVE_DIR"
+q_model.save_pretrained(saved_dir)
+
+# load quant model
+loaded_model = AutoModelForCausalLM.from_pretrained(saved_dir)
+```
+
+## Usage For Intel GPU
+Intel® Neural Compressor implement weight-only quantization for intel GPU(PVC/ARC/MTL) with [Intel-extension-for-pytorch](https://github.com/intel/intel-extension-for-pytorch).
```
Comment (thuang6) [inline docs/source/3x/transformers_like_api.md:125]: 1. implement -> implements
2. intel GPU -> Intel GPU, 
3. add LNL to GPU list
4. Intel-extension-for-pytorch ->  intel-extension-for-pytorch

File: docs/source/3x/transformers_like_api.md [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+Now 4-bit/8-bit inference with `RtnConfig`, `GPTQConfig`, `AutoRoundConfig` are support on intel GPU device.
+
+We support experimental woq inference on intel GPU(PVC/ARC/MTL) with replacing Linear op in PyTorch. Validated models: Qwen-7B, Llama-7B, Phi-3.
```
Comment (thuang6) [inline docs/source/3x/transformers_like_api.md:129]: 1. intel GPU -> Intel GPU
2. add LNL to GPU list
3. Validated models should use full name, meta-llama/Meta-Llama-3-8B, meta/llama-Llama-2-7b-hf, Qwen/Qwen-7B-Chat, microsoft/Phi-3-mini-4k-instruct

File: docs/source/3x/transformers_like_api.md [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+
+We support experimental woq inference on intel GPU(PVC/ARC/MTL) with replacing Linear op in PyTorch. Validated models: Qwen-7B, Llama-7B, Phi-3.  
+
+Here are the example codes.
+
+#### Prepare Dependency Packages
+1. Install Oneapi Package  
+The Oneapi DPCPP compiler is required to compile intel-extension-for-pytorch. Please follow [the link](https://www.intel.com/content/www/us/en/developer/articles/guide/installation-guide-for-oneapi-toolkits.html) to install the OneAPI to "/opt/intel folder".
+
+2. Build and Install PyTorch and Intel-extension-for-pytorch
```
Comment (thuang6) [inline docs/source/3x/transformers_like_api.md:137]: IPEX has different installation procedure for different device, e.g. PVC is different with MTL or LNL.  we should not put detail IPEX installation info in our doc,  just need a link to IPEX-XPU installation guide doc

File: docs/source/3x/transformers_like_api.md [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+export OCL_ICD_VENDORS=/etc/OpenCL/vendors
+export CCL_ROOT=${CONDA_PREFIX}
+source /opt/intel/oneapi/setvars.sh --force
+export LLM_ACC_TEST=1
+pip install -r requirements.txt
+
+python setup.py install
+```
+
+3. Install Neural-compressor
```
Comment (thuang6) [inline docs/source/3x/transformers_like_api.md:157]: This is inside INC doc,  no need for install INC section

Comment (Kaihui-intel) [reply docs/source/3x/transformers_like_api.md]: removed, lll comments have been resolved

**Review by thuang6 — APPROVED** [review]

**Review by XuehaoSun — APPROVED** [review]

**Pull request merged**